src/api/solr-client.js

```diff
@@ -62,10 +62,16 @@ class SolrClient {
     };
 
     this.sendQuery(queryReducer(this.state.query, payload));
-
     return this;
   }
 
+  // Function to set up new start and new Fields, when the page is loaded at the first time
+  // this is another initialize function with searchFields and start paramters
+  setInitPage(start, newFields, newSortFields) {
+    const payload = {type: "SET_INIT_STATE", newFields: newFields, newSortFields: newSortFields, newStart: start};
+    this.sendQuery(queryReducer(this.state.query, payload));
+  }
+
```

Comment:
New function to initialize the solrClient with URL parameters

Reply:
Per @datadavev's suggestion, could we maybe put this new function in a different file so we can easily track our changes?  I don't know if it's possible to add methods to one class using multiple files in solr.

Reply:
Yeah. I am thinking about a new way without modifying the original library.

Reply:
One option might be to subclass the built-in library, put all our new methods in the subclass, and use the subclass in our application code.

Reply:
I noticed there is another initialized function. Hope it can help us so we don't need to use the customized functions.